modules/core/src/main/java/org/apache/ignite/internal/processors/affinity/GridAffinityAssignmentCache.java

```diff
@@ -304,6 +309,60 @@ public void onReconnected() {
     }
 
     /**
+     * @param assignment List indexed by partition number.
+     * @param nodesCnt Nodes count.
+     */
+    private void printDistribution(List<List<ClusterNode>> assignment, int nodesCnt) {
+        Float ignitePartDistribution = getFloat(IGNITE_PART_DISTRIBUTION_WARN_THRESHOLD, 0.1f);
+
+        int[] partitionsByLocalNode = new int[nodesCnt];
+
+        int totalPrimaryCnt = 0;
+
+        int totalBackupCnt = 0;
+
+        for (int i = 0; i < nodesCnt; i++) {
+            for (List<ClusterNode> partitionByNodes : assignment) {
+                if (partitionByNodes.get(i).isLocal())
```

Comment:
boolean can only be true or false because it's a primitive datatype
```
   /** Grid local node flag (transient). */
    private boolean loc;
```